mscore/musescore.cpp

```diff
@@ -5382,8 +5382,10 @@ void MuseScore::cmd(QAction* a)
 
 void MuseScore::endCmd()
       {
-      if (timeline())
-            timeline()->updateGrid();
+      if (timeline()) {
+            if (timeline()->isVisible())
+                  timeline()->updateGrid();
```

Comment:
Well, for your first point, as soon as the timeline is made visible, it will update itself. I think your other idea, of putting the code to check if the timeline is visible inside the timeline class, is a good idea. I'll reimplement it like that, then we don't have to worry about update calls from elsewhere in the code.